src/sensors/softfusion/drivers/bmi270.h

```diff
@@ -405,10 +405,11 @@ struct BMI270 {
 				}
 				getFromFifo<uint8_t>(i, read_buffer);  // skip 1 byte
 			} else if ((header & Fifo::ModeMask) == Fifo::DataFrame) {
-				const uint8_t required_length
-					= (((header & Fifo::GyrDataBit) >> Fifo::GyrDataBit)
-					   + ((header & Fifo::AccelDataBit) >> Fifo::AccelDataBit))
-					* 6;
+				uint8_t gyro_data_length
+					= (header & Fifo::GyrDataBit) == Fifo::GyrDataBit ? 6 : 0;
```

Comment:
it's enough to (header & Fifo::GyrDataBit ? 6 : 0) afaik

similar for accel

Reply:
Fixed!